reconcile/utils/gql.py

```diff
@@ -198,13 +193,39 @@ def get_resource(path: str) -> dict[str, Any]:
     return get_api().get_resource(path)
 
 
+class PersistentRequestsHTTPTransport(RequestsHTTPTransport):
+    def connect(self):
+        if self.session is None:
+            # Copied over from RequestsHTTPTransport
+            self.session = requests.Session()
+            # we did not implement this in our copy!
+            assert self.retries == 0
+
+    def close(self) -> None:
+        pass
```

Comment:
how are we closing in case the program ends?

Reply:
TCP connections will just time out....

Reply:
what about closing them via `__del__` method?

Reply:
Just checked, we have other places where we use Sessions without caring for threads and do not clean them up... 
I'll follow up on that

Reply:
Added _SessionClosingCache_ that should take care of closing sessions